src/system/SystemPacketBuffer.h

```diff
@@ -206,29 +206,29 @@ class DLL_EXPORT PacketBuffer : private pbuf
      *  @param[in,out] aChainHead - the head of the buffer chain the current buffer belongs to.  May be \c nullptr if the current
      *      buffer is the head of the buffer chain.
      */
-    void SetDataLength(uint16_t aNewLen, const PacketBufferHandle & aChainHead);
-    void SetDataLength(uint16_t aNewLen) { SetDataLength(aNewLen, nullptr); }
+    void SetDataLength(size_t aNewLen, const PacketBufferHandle & aChainHead);
+    void SetDataLength(size_t aNewLen) { SetDataLength(aNewLen, nullptr); }
```

Comment:
For lwIP, looks like we can't take in > 16 bits. There needs to be range checks.

Reply:
SetDataLength() implementation now has a max check for aNewLen against UNIT16_MAX for LwIP.